portfolio/src/main/java/com/google/sps/servlets/DoggoRedditServlet.java

```diff
@@ -0,0 +1,136 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+import java.io.PrintWriter;
+
+@WebServlet("/doggo-servlet")
+public final class DoggoRedditServlet extends HttpServlet {
+
+  final String COMMENT_TEXT = "commentText";
+  final String TIMESTAMP = "timestamp";
+  final String IMAGE_URL = "imageUrl";
+  final String FEED_POST = "FeedPost";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Entity postEntity = new Entity(FEED_POST);
+    long timestamp = System.currentTimeMillis();
+    String uploadURL = getUploadedFileUrl(request, "image");
+
+    postEntity.setProperty(COMMENT_TEXT, request.getParameter(COMMENT_TEXT));
+    postEntity.setProperty(TIMESTAMP, timestamp);
+    postEntity.setProperty(IMAGE_URL, uploadURL);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(postEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query(FEED_POST).addSort(TIMESTAMP, SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<FeedPost> posts = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      
+      String commentText = (String) entity.getProperty(COMMENT_TEXT);
+      String imageUrl = (String) entity.getProperty(IMAGE_URL);
+      long timestamp = (long) entity.getProperty(TIMESTAMP);
+      long id = entity.getKey().getId();
+
+      FeedPost post = new FeedPost(commentText, imageUrl, timestamp);
+      posts.add(post);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(posts));
+  }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```

Comment:
maybe rename `uploadFileAndReturnUrl`?

Reply:
technically this function doesn't upload the file right? it only retrieves the image url from blobstore. the uploading was done back in [BlobstoreUploadServlet.java](https://github.com/jirehcwe/google-software-product-sprint/pull/3/commits/0afa226f276e2f00d9ce8ee74b5fefa07404a52b).

Reply:
Ahh I misunderstood haha. Thanks for clarifying